tests/conftest.py

```diff
@@ -43,6 +47,37 @@ def coordinator_service():
     server.stop(0)
 
 
+@pytest.fixture
+def mock_coordinator_service():
+    """[summary]
+
+    .. todo:: Advance docstrings (https://xainag.atlassian.net/browse/XP-425)
+    """
+
+    server = grpc.server(futures.ThreadPoolExecutor(max_workers=1))
+    agg = ModelSumAggregator()
+    ctrl = IdController()
+    coordinator = Coordinator(
+        num_rounds=2,
+        minimum_participants_in_round=1,
+        fraction_of_participants=1.0,
+        aggregator=agg,
+        controller=ctrl,
+    )
+    coordinator_grpc = CoordinatorGrpc(coordinator)
+    coordinator_pb2_grpc.add_CoordinatorServicer_to_server(coordinator_grpc, server)
+    server.add_insecure_port("localhost:50051")
+    server.start()
+    terminate_event = threading.Event()
+    monitor_thread = threading.Thread(
```

Comment:
Maybe we need to `join()` this thread, just to make sure it terminated?

Reply:
yes, to confirm the expected response of setting `terminate_event`.